pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/ProtoParserToolsTest.java

```diff
@@ -196,7 +196,7 @@ void testReadString_maxSize() throws IOException {
 
     @Test
     void testReadString_incomplete() throws IOException {
-        final int length = rng.nextInt(0, 100);
+        final int length = rng.nextInt(1, 100);
```

Comment:
This is a huge PR. And this lowers the quality of code review that can be provided, leading to potential hidden bugs.

You could've split it into multiple small manageable parts, for example:
1. Upgrade to Java 21
2. Update the checkstyle and similar configs
3. Introduce documentation
4. Introduce small utilities
5. Introduce core interfaces/implementations (perhaps still split even further)
6. Introduce a test service

Yes, this would require that many PRs. However, the quality of reviews, and as a result, the quality of the code could be improved this way. Also, small PRs should (in theory) lead to faster turn around times.

(Yes, a huge PR can also get a very fast `Approved`, with a caveat that nothing has actually been reviewed properly or at all. And this is not the right thing to do.)

Reply:
Ah, it is actually not that big, but the PR has a bunch of diffs that it shouldn't. Time for some git magic.

Reply:
Ok, that should be better, all the noise is gone now. I still don't know why github does that, because if I do a diff from the branch (before I "fixed" it) and main, it shows me only these files as different. But the PR shows a ton more changes, even though they aren't actually different from main. I'm not sure why it does that, but I've got it happy now (although I had to force-push to the branch. Blah).

I could factor out the change to Java 21 into another PR, but that's almost the only change that isn't actually related to the helidon project now.

Reply:
Prepare for a deluge of small PRs 😄 😄

Reply:
https://github.com/hashgraph/pbj/pull/264

Reply:
https://github.com/hashgraph/pbj/pull/265

Reply:
https://github.com/hashgraph/pbj/pull/266